cmd/cerbos/healthcheck/healthcheck.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2021-2022 Zenauth Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package healthcheck
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/alecthomas/kong"
+	svcv1 "github.com/cerbos/cerbos/api/genpb/cerbos/svc/v1"
+	"github.com/cerbos/cerbos/internal/util"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/local"
+	healthpb "google.golang.org/grpc/health/grpc_health_v1"
+)
+
+const (
+	defaultGRPCHostPort = "127.0.0.1:3593"
+	defaultHTTPHostPort = "127.0.0.1:3592"
+
+	help = `
+Performs a healthcheck on a Cerbos PDP. This can be used as a Docker HEALTHCHECK command.
+By default, the local Cerbos gRPC endpoint (127.0.0.1:3593) will be checked using the gRPC healthcheck protocol. This is usually sufficient for most cases as the Cerbos REST API is built on top of the gRPC API as well.   
+
+Examples:
+
+# Check gRPC endpoint
+
+cerbos healthcheck
+
+# Check gRPC endpoint and ignore server certificate verification
+
+cerbos healthcheck --insecure
+
+# Check https endpoint, ignoring server certificate verification and with a custom timeout
+
+cerbos healthcheck --kind=https --insecure --timeout=2s
+
+# Use a different host address
+
+cerbos healthcheck --kind=http --host-port=10.0.1.5:3592
+`
+)
+
+type Cmd struct {
+	Kind      string        `help:"Healthcheck kind (${enum})" default:"grpc" enum:"grpc,http,https" env:"CERBOS_HC_KIND"`
+	HostPort  string        `help:"Host and port to check. Defaults to 127.0.0.1:3592 for http/https and 127.0.0.1:3593 for grpc" env:"CERBOS_HC_HOST"`
+	CACert    string        `help:"CA Certificate to use to verify the server certificate" type:"existingfile"`
+	Insecure  bool          `help:"Do not verify server certificate" default:"false"`
+	Plaintext bool          `help:"No TLS (gRPC only)" default:"false"`
+	Timeout   time.Duration `help:"Healthcheck timeout" default:"10s" env:"CERBOS_HC_TIMEOUT"`
```

Comment:
I personally wouldn't automatically associate "HC" with "healthcheck", is it widely adopted? My preference would be to use `CERBOS_HEALTHCHECK_` as the prefix for the environment variables.

Reply:
I have seen HC being used to refer to the healthcheck in some places. But, mainly, I just want to keep the environment variable name short.

Reply:
I'd imagine it would typically be used in things like Kubernetes config files rather than typed at the command line, so I would favour clarity over brevity. But there's an argument either way so if you'd rather keep it as `HC`, no worries.